bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -2589,6 +2589,9 @@ private static boolean checkIsLikeRecordType(Object sourceValue, BRecordType tar
                 addErrorMessage((errors == null) ? 0 : errors.size(), "missing required field '" + fieldNameLong +
                         "' of type '" + targetField.getFieldType().toString() + "' in record '" + targetType + "'",
                         errors);
+                if ((errors == null) || (errors.size() == MAX_TYPECAST_ERROR_COUNT + 1)) {
```

Comment:
errors.size() >= MAX_TYPECAST_ERROR_COUNT should be safe i think instead of checking exact equal.